### PR TITLE
Added support for building LMDB v0.9.25

### PR DESCRIPTION
**Pull request by blenessy**
The LMDB surces are maintained here:
https://git.openldap.org/openldap/openldap

### CODE REVIEW

**Comment by zalan-svc** [comment]
Sure @giordano , sounds good to me.

**Comment by blenessy** [comment]
Hold on merging please (I've found a segfault I'm investigating)... I will re-open when fixed.

**Comment by giordano** [comment]
Oh no! :disappointed:

**Comment by blenessy** [comment]
@giordano sorry about that ... I've found and fixed the problem and confirmed that both the MacOS (64-bit) lib no longer segfaults. (Linux was working before)

File: L/LMDB/build_tarballs.jl [hunk 1]
```diff
@@ -5,17 +5,21 @@ version = v"0.9.25"
...
 sources = [
-    ArchiveSource("https://git.openldap.org/openldap/openldap/-/archive/LMDB_0.9.25/openldap-LMDB_$(version).tar.gz",
+    ArchiveSource("https://git.openldap.org/openldap/openldap/-/archive/LMDB_$(version)/openldap-LMDB_$(version).tar.gz",
                   "4f6eebe5ad98c10a75badd106f498ee2249d454352d048c78a49c99c940d4cae"),
 ]
 
 # Bash recipe for building across all platforms
-# rm: remove man files (it does not name sense)
-# exeext: Makefile does not support extensions - need to rename execuables manually
+# - CC: mdb_env_close0 segfaults if gcc is used instead of clang on Mac
```
Comment (giordano) [inline L/LMDB/build_tarballs.jl:13]: I'm not sure I understand this comment: the default compiler is already `clang` for macOS and FreeBSD

Comment (blenessy) [reply L/LMDB/build_tarballs.jl]: `env` says `CC=cc` in the BinaryBuilder container for the MacOS target. I thought that was different from `clang`. Thanks to your comment, I've revisited this and confirmed that `cc` is in fact `clang`, so I can simplify the fix.

The root cause is [this line](https://git.openldap.org/openldap/openldap/-/blob/LMDB_0.9.25/libraries/liblmdb/Makefile#L21) which picks `gcc` for all targets (incl. MacOS). `make CC=$(CC) ...` should be enough to fix this.

Comment (giordano) [reply L/LMDB/build_tarballs.jl]: > `make CC=$(CC) ...` should be enough to fix this.

I think so, thanks!

**Pull request merged**